seisbench/generate/augmentation.py

```diff
@@ -670,3 +672,194 @@ def __call__(self, state_dict):
+            raise ValueError(msg)
+
+    def __call__(self, state_dict) -> np.array:
+        x, metadata = state_dict[self.key[0]]
+
+        if self.key[0] != self.key[1]:
+            # Ensure metadata is not modified inplace unless input and output key are anyhow identical
+            metadata = copy.deepcopy(metadata)
+
+        # Create random angle alpha if alpha is not given in init
+        if self.alpha is None:
+            alpha = np.random.uniform(-np.pi, np.pi)
+        else:
+            alpha = self.alpha
+
+        trace_components = metadata.get("trace_component_order")
+        if not trace_components:
+            msg = "Keyword 'trace_component_order' is missing in metadata, therefore rotation is not possible."
+            raise ValueError(msg)
+
+        data_comp1 = copy.copy(x[trace_components.index(self.components[0]), :])
+        data_comp2 = copy.copy(x[trace_components.index(self.components[1]), :])
+
+        # Rotate by angle alpha
+        x[trace_components.index(self.components[0]), :] = data_comp1 * np.cos(
+            alpha
+        ) - data_comp2 * np.sin(alpha)
+        x[trace_components.index(self.components[1]), :] = data_comp1 * np.sin(
+            alpha
+        ) + data_comp2 * np.cos(alpha)
+
+        state_dict[self.key[1]] = (x, metadata)
+
+    def __str__(self):
+        if not self.alpha:
+            return f"Rotation of {self.components} components by arbitrary angle"
+        else:
+            return f"Rotation of {self.components} components by {np.round(self.alpha, 2)} rad"
+
+
+class RealNoise:
+    """
+    Adds recorded noise from an extra noise data set to the data.
+    This method was inspired by the following publication:
+    Jun Zhu, Lihua Fang, Fajun Miao, Liping Fan, Ji Zhang, Zefeng Li
+    Deep learning and transfer learning of earthquake and quarry-blast discrimination: applications to southern
+    California and eastern Kentucky
+    Geophysical Journal International, Volume 236, Issue 2, February 2024, Pages 979–993,
+    https://doi.org/10.1093/gji/ggad463
+
+    :param noise_dataset: WaveformDataset that only contains noise traces
+    :type noise_dataset: seisbench.data.base.WaveformDataset
+    :param scale: Tuple of minimum and maximum relative amplitude of the noise.
+                  Relative amplitude is defined either by the absolute maximum or the root-mean-square of input array
+                  (see scaling_type). Default value is 0.5
+    :type scale: tuple(float, float)
+    :param key: The keys for reading from and writing to the state dict.
+                If key is a single string, the corresponding entry in state dict is modified.
+                Otherwise, a 2-tuple is expected, with the first string indicating the key to
+                read from and the second one the key to write to.
+    :type key: str, tuple[str, str]
+    :param probability: Probability of adding noise samples to data set, e.g. if probability=0.5, 50% of all samples
```

Comment:
I don't think this is necessary, because there is the `OneOf` augmentation. Maybe we could remove this parameter here and instead explain that only adding noise to some of the examples can be achieved using `OneOf`?